yaaredis/pool.py

```diff
@@ -647,3 +648,235 @@ def get_node_by_slot(self, slot):
         if self.readonly:
             return random.choice(self.nodes.slots[slot])
         return self.get_master_node_by_slot(slot)
+
+
+class BlockingClusterConnectionPool(ConnectionPool):
```

Comment:
What do you think about inheriting from `ClusterConnectionPool` instead? Looks like a bunch of stuff got reimplemented as a direct copy here -- we can probably cut down on a bunch of the complexity if we inherit that behaviour instead.

Hopefully, that'll make the differences a bit more obvious as well -- even with you're warning I completely missed the difference in `asserts` between the two classes.